Libplanet/KeyStore/Web3KeyStore.cs

```diff
@@ -0,0 +1,185 @@
+            ListFiles().Select(pair => Tuple.Create(pair.Item1, Get(pair.Item2)));
+
+        /// <inheritdoc/>
+        public IEnumerable<Guid> ListIds() =>
+            ListFiles().Select(pair => pair.Item1);
+
+        /// <inheritdoc/>
+        public ProtectedPrivateKey Get(Guid id)
+        {
+            string name;
+            try
+            {
+                (_, name) = ListFiles().First(pair => pair.Item1.Equals(id));
+            }
+            catch (InvalidOperationException)
+            {
+                throw new NoKeyException(id, "There is no key with such ID");
+            }
+
+            return Get(name);
+        }
+
+        /// <inheritdoc/>
+        public Guid Add(ProtectedPrivateKey key)
+        {
+            if (key is null)
+            {
+                throw new ArgumentNullException(nameof(key));
+            }
+
+            Guid keyId = Guid.NewGuid();
+            string filename = string.Format(
+                CultureInfo.InvariantCulture,
+                NameFormat,
+                DateTimeOffset.UtcNow,
+                keyId
+            );
+            var keyPath = System.IO.Path.Combine(Path, filename);
+            using Stream f = new FileStream(keyPath, FileMode.CreateNew);
+            key.WriteJson(f, keyId);
+            return keyId;
+        }
+
+        /// <inheritdoc/>
+        public void Remove(Guid id)
+        {
+            foreach ((Guid keyId, string keyPath) in ListFiles())
+            {
+                if (keyId.Equals(id))
+                {
+                    System.IO.File.Delete(keyPath);
+                    return;
+                }
+            }
+
+            throw new NoKeyException(id, "No key have such ID");
+        }
+
+        private IEnumerable<(Guid, string)> ListFiles()
+        {
+            IEnumerable<string> keyPaths = Directory.EnumerateFiles(Path);
+            foreach (string keyPath in keyPaths)
+            {
+                if (System.IO.Path.GetFileName(keyPath) is string f)
+                {
+                    Match m = NamePattern.Match(f);
+                    if (m.Success)
+                    {
+                        Guid id;
+                        try
+                        {
+                            id = Guid.Parse(m.Groups[1].Value);
+                        }
+                        catch (FormatException e)
+                        {
+                            _logger.Debug(e, "Failed to parse the file name: {keyPath}");
+                            continue;
+                        }
+
+                        yield return (id, keyPath);
+                    }
+                }
+            }
+        }
+
+        private ProtectedPrivateKey Get(string name)
+        {
+            using (StreamReader reader = new StreamReader(System.IO.Path.Combine(Path, name)))
+            {
+                return ProtectedPrivateKey.FromJson(reader.ReadToEnd());
+            }
```

Comment:
Could we use `using var` syntax here?

```suggestion
            using var reader = new StreamReader(System.IO.Path.Combine(Path, name)))
            return ProtectedPrivateKey.FromJson(reader.ReadToEnd());
```

Reply:
Due to [Sonar Analyzer's bug][1], that syntax cause a false alarm.  Anyway that bug was fixed in the [version 8.3], but again it introduces another bug…  😇  I hope the next version of Sonar Analyzer to eliminate all of these bugs.

[1]: https://community.sonarsource.com/t/dispose-with-using-triggers-s2930-c-8-0/13740
[2]: https://github.com/SonarSource/sonar-dotnet/releases/tag/8.3.0.14607